### DISCUSSION TITLE
No child spans created for subresources in jetty and jersey

### DISCUSSION THREAD

**Discussion by harveychong** [open-telemetry/opentelemetry-java-instrumentation#9646] [Q&A]
I am new to OpenTelemetry.

I have two resources `A` and `B` and `B` is a sub-resource of `A.`  When the sub-resource `B` was called, I found  Parent ID of both resources are same (I am using ZipKin).  I expect Parent ID of the sub-resource `B` should be Span ID of resource `A`.  I am using Jetty and Jersey (JDK 17).

In addition, when I applied my own extension (developed by myself) to my application entry point (a java method) where a postgresql database is called, I found Parent ID of the database calls is none but I expect it should be child of my application.

I am not sure what I did is wrong.  Please help me for it.  

Many thanks in advance.

**Comment by laurit** [comment]
This question is a better fit for the `opentelemetry-java-instrumentation` project.
Typically there is a parent child relationship between span when one operation happens inside the other. For jax-rs sub resource the parent returns an instance of the sub resource, it does not call the actual resource method, hence the parent and sub resource will appear as siblings.
You'll have to provide a minimal sample app that reproduces the second issue. If your code or the sql library you use dispatch the querying to a different thread instrumentation may be needed to make these spans appear in the same trace as caller.
